core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java

```diff
@@ -165,6 +165,9 @@
     private Supplier<DynamicFiltersStats> dynamicFiltersStatsSupplier = () -> DynamicFiltersStats.EMPTY;
     private final Object dynamicFiltersStatsSupplierLock = new Object();
 
+    private final AtomicBoolean committed = new AtomicBoolean();
```

Comment:
`committed` is confusing - you think transactions - while this is not related here at all.
Maybe lets have more straightforward names here:
 * resultsReady
 * resultsConsumed
?

Reply:
> committed is confusing - you think transactions - while this is not related here at all.

It is actually about transactions. Basically before considering a query as `FINISHED` it has to finish active execution (be in the `FINISHING` state), commit a transaction if one is active and auto commit is enabled (`commited`) and in addition to that with this PR it would be required for all the results to be consumed (`resultsConsumed`).